src/averageAges.js

```diff
@@ -20,6 +20,16 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  const men = people.filter(person => person.sex === 'm');
+  const menLifespans = century
+    ? men.filter(person => Math.ceil(person.died / 100) === century)
+      .map(person => person.died - person.born)
+    : men.map(person => person.died - person.born);
+  const totalMenLifespan = menLifespans.reduce(
+    (sum, lifespan) => sum + lifespan, 0);
+  const averageMenLifespan = totalMenLifespan / menLifespans.length;
+
+  return averageMenLifespan;
```

Comment:
move logic related to calculate average age to the separate function and reuse it